### PR TITLE
fix Llama models messages to prompt conversion for AWS Bedrock

### LINKED ISSUE #3297: [Bug]: Gibberish output of Llama-3 models on AWS Bedrock
### What happened?

Llama-3 models are not working properly on AWS Bedrock right now and output gibberish. Probably because the instruct models now follow a different template.

```python
response = client.chat.completions.create(
    model="llama-3-8b", 
    messages = [
        {
            "role": "user",
            "content": "hi"
        }
    ],
    max_tokens=16
)

print(response.choices[0].message.content)
```

### Relevant log output

```shell
K, et al. (2014). "A systematic review of the effectiveness
```

### Twitter / LinkedIn details

_No response_

### PR DESCRIPTION
**Pull request by aswny**
Fix: #3297

----
@krrishdholakia I saw that you added support for Llama-3 recently in commit [df7db2b](https://github.com/BerriAI/litellm/commit/df7db2b870d2e1201888bb625c446e4473759ffb) but that didn't cover Llama Bedrock models.

Please let me know if this looks good otherwise.

### CODE REVIEW

File: litellm/llms/prompt_templates/factory.py [hunk 1]
```diff
@@ -1346,6 +1346,16 @@ def prompt_factory(
                 return anthropic_pt(messages=messages)
         elif "mistral." in model:
             return mistral_instruct_pt(messages=messages)
+        elif "llama2" in model:
+            return llama_2_chat_pt(messages=messages)
+        elif "llama3" in model:
```
Comment (ghost) [inline litellm/llms/prompt_templates/factory.py]: can you check if it's an 'instruct' model before applying the chat template (same for llama2).

Comment (aswny) [reply litellm/llms/prompt_templates/factory.py]: Bedrock only supports Llama-2-Chat and Llama-3-Instruct models [1].

Nonetheless, I think it still makes sense to add a check just to be future proof. Will update.

 [1] https://docs.aws.amazon.com/bedrock/latest/userguide/model-ids.html#model-ids-arns

**Comment by aswny** [comment]
I did a quick check using latest commit too and it seems to be working now.

Screenshot 2024-04-25 at 23 15 33

**Comment by ghost** [comment]
@aswny do you mean the latest commit on main? or your PR branch

File: litellm/llms/prompt_templates/factory.py (hunk 1 above)
Comment (ghost) [reply litellm/llms/prompt_templates/factory.py]: i see non-chat llama2 models on bedrock

Screenshot 2024-04-25 at 11 20 10 AM

**Comment by aswny** [comment]
I mean this PR branch.

**Comment by ghost** [comment]
Just reviewed. LGTM!

**Pull request merged**